examples/Tests/ParametricEq.hpp

```diff
@@ -0,0 +1,585 @@
+#pragma once
+
+#include "EQDiagram.hpp"
+#include "ParametricEqControls.hpp"
+
+#include <DspFilters/Butterworth.h>
+#include <DspFilters/ChebyshevI.h>
+#include <DspFilters/Filter.h>
+#include <halp/custom_widgets.hpp>
+#include <halp/layout.hpp>
+
+#include <iostream>
+#include <map>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include <string_view>
+
+namespace examples
+{
+// ********************************************************************************
+//                                  Actual EQ
+// ********************************************************************************
+struct ParametricEq
+{
+  static constexpr auto name() { return "Parametric Equalizer"; }
+  static constexpr auto c_name() { return "ngry_parameq"; }
+  static constexpr auto author() { return "Nicolas Gry"; }
+  static constexpr auto category() { return "Demo"; }
+  static constexpr auto description()
+  {
+    return "Parametric Equalizer, using Butterworth equations for all filters except "
+           "the bandshelf filter (which is computed with Chebyshev Type I equations).";
+  }
+  static constexpr auto uuid() { return "ec972d12-edba-4a9c-9736-9176efad17e3"; }
+
+  static constexpr int bandsNb = 8;  // EQ Bands number
+  static constexpr int maxOrder = 7; //Filters maximum order
+  struct ins
+  {
+    struct
+    {
+      static constexpr auto name() { return "Input"; }
+      const double** samples;
+      int channels;
+      float sampleRate;
+    } audio;
+
+    EQBand band1, band2, band3, band4, band5, band6, band7, band8;
+
+    // Common filters properties
+    filterToggle toggle1, toggle2, toggle3, toggle4, toggle5, toggle6, toggle7, toggle8;
+    order<maxOrder> order1, order2, order3, order4, order5, order6, order7, order8;
+
+    // Specific filters properties (not all filters require them)
+    cutoffFrequency cutoffFreq1, cutoffFreq2, cutoffFreq3, cutoffFreq4, cutoffFreq5,
+        cutoffFreq6, cutoffFreq7, cutoffFreq8;
+    centerFrequency centerFreq1, centerFreq2, centerFreq3, centerFreq4, centerFreq5,
+        centerFreq6, centerFreq7, centerFreq8;
+    gain gain1, gain2, gain3, gain4, gain5, gain6, gain7, gain8;
+    frequencyBandWidth freqBandWidth1, freqBandWidth2, freqBandWidth3, freqBandWidth4,
+        freqBandWidth5, freqBandWidth6, freqBandWidth7, freqBandWidth8;
+    passbandRipple passbandRipple1, passbandRipple2, passbandRipple3, passbandRipple4,
+        passbandRipple5, passbandRipple6, passbandRipple7, passbandRipple8;
+
+  } inputs;
+
+  //                                      Available filters
+  static constexpr auto chans = 2;
+  Dsp::Filter * lowpassFilter, * highpassFilter, * bandpassFilter, * bandstopFilter, * lowshelfFilter, * highshelfFilter, * bandshelfFilter;
+
+  // EQ Bode diagram calculator
+  static constexpr auto plotPoints = 5000;
+  EQDiagramCalculator<plotPoints, bandsNb> diagramCalculator;
+  bool  toRecompute;
```

Comment:
initialize

Reply:
do you mean to put a controller on it in the ui ? so the user can choose the value?